### PR TITLE
[admin] Use OTEL helper instead of tlscfg

### PR DESCRIPTION
**Pull request by chahatsagarmain**
## Which problem is this PR solving?
- Part of #4316

## Description of the changes
- 

## How was this change tested?
- 

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: cmd/internal/flags/admin.go [hunk 1]
```diff
@@ -35,13 +34,12 @@ var tlsAdminHTTPFlagsConfig = tlscfg.ServerFlagsConfig{
...
-	mux                  *http.ServeMux
-	server               *http.Server
-	tlsCfg               *tls.Config
-	tlsCertWatcherCloser io.Closer
+	logger        *zap.Logger
+	adminHostPort string
+	hc            *healthcheck.HealthCheck
+	mux           *http.ServeMux
+	server        *http.Server
+	tlsCfg        *tls.Config
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:42]: this and adminHostPort should be replaced by confighttp.ServerConfig

File: cmd/internal/flags/admin.go [hunk 2]
```diff
@@ -101,7 +97,7 @@ func (s *AdminServer) Handle(path string, handler http.Handler) {
 
 // Serve starts HTTP server.
 func (s *AdminServer) Serve() error {
-	l, err := net.Listen("tcp", s.adminHostPort)
+	l, err := net.Listen("tcp", s.serverCfg.Endpoint)
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:100]: we should be using `confighttp.ServerConfig` functions now (ToLister, ToServer)

**Comment by chahatsagarmain** [comment]
@yurishkuro `--- FAIL: TestBuilderWithCollectors (0.00s)
    collector_proxy_test.go:73: 
        	Error Trace:	/home/runner/work/jaeger/jaeger/cmd/agent/app/reporter/grpc/collector_proxy_test.go:73
        	            				/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/runtime/asm_amd64.s:1700
        	Error:      	Received unexpected error:
        	            	grpc: the server has been stopped
        	Test:       	TestBuilderWithCollectors`

File: cmd/internal/flags/admin.go [hunk 3]
```diff
@@ -35,22 +34,22 @@ var tlsAdminHTTPFlagsConfig = tlscfg.ServerFlagsConfig{
...
-	hc                   *healthcheck.HealthCheck
-	mux                  *http.ServeMux
-	server               *http.Server
-	tlsCfg               *tls.Config
-	tlsCertWatcherCloser io.Closer
+	logger    *zap.Logger
+	hc        *healthcheck.HealthCheck
+	mux       *http.ServeMux
+	server    *http.Server
+	serverCfg *confighttp.ServerConfig
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:41]: doesn't need to be a pointer, it's not optional

File: cmd/internal/flags/admin.go [hunk 4]
```diff
@@ -67,29 +66,26 @@ func (s *AdminServer) setLogger(logger *zap.Logger) {
...
 
 // InitFromViper initializes the server with properties retrieved from Viper.
 func (s *AdminServer) initFromViper(v *viper.Viper, logger *zap.Logger) error {
 	s.setLogger(logger)
 
-	s.adminHostPort = v.GetString(adminHTTPHostPort)
-	var tlsAdminHTTP tlscfg.Options
+	s.serverCfg.Endpoint = v.GetString(adminHTTPHostPort)
 	tlsAdminHTTP, err := tlsAdminHTTPFlagsConfig.InitFromViper(v)
+	tlsAdminHTTPConfig := tlsAdminHTTP.ToOtelServerConfig()
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:79]:
```suggestion
```

File: cmd/internal/flags/admin.go [hunk 5]
```diff
@@ -67,29 +66,26 @@ func (s *AdminServer) setLogger(logger *zap.Logger) {
...
 	if err != nil {
 		return fmt.Errorf("failed to parse admin server TLS options: %w", err)
 	}
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:82]:
```suggestion
	if err != nil {
		return fmt.Errorf("failed to parse admin server TLS options: %w", err)
	}
	tlsAdminHTTPConfig := tlsAdminHTTP.ToOtelServerConfig()
```

File: cmd/internal/flags/admin.go [hunk 6]
```diff
@@ -67,29 +66,26 @@ func (s *AdminServer) setLogger(logger *zap.Logger) {
...
-	if tlsAdminHTTP.Enabled {
-		tlsCfg, err := tlsAdminHTTP.Config(s.logger) // This checks if the certificates are correctly provided
+	if tlsAdminHTTPConfig != nil {
+		s.serverCfg.TLSSetting = tlsAdminHTTPConfig
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:84]: this is the only thing that needs to happen, everything else is handled by ToListener

File: cmd/internal/flags/admin.go [hunk 7]
```diff
@@ -115,34 +113,38 @@ func (s *AdminServer) Serve() error {
...
 	}
-	if s.tlsCfg != nil {
-		s.server.TLSConfig = s.tlsCfg
+	s.server.ErrorLog = errorLog
+
+	if s.serverCfg.TLSSetting != nil {
+		tlsConfig, err := s.serverCfg.TLSSetting.LoadTLSConfig(context.Background())
+		if err != nil {
+			s.logger.Error("failed to load tls config", zap.Error(err))
+			s.hc.Set(healthcheck.Broken)
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:134]: why is this needed?

File: cmd/internal/flags/admin.go [hunk 8]
```diff
@@ -115,34 +113,38 @@ func (s *AdminServer) Serve() error {
...
-	s.logger.Info("Starting admin HTTP server", zap.String("http-addr", s.adminHostPort))
+	s.logger.Info("Starting admin HTTP server", zap.String("http-addr", s.serverCfg.Endpoint))
 	go func() {
-		var err error
-		if s.tlsCfg != nil {
-			err = s.server.ServeTLS(l, "", "")
-		} else {
-			err = s.server.Serve(l)
-		}
+		err := s.server.Serve(l)
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:140]: might want to use a waitgroup here to block return at L147 until the listen goroutine has started

File: cmd/internal/flags/admin.go [hunk 9]
```diff
@@ -67,29 +67,26 @@ func (s *AdminServer) setLogger(logger *zap.Logger) {
...
 	}
-	if tlsAdminHTTP.Enabled {
-		tlsCfg, err := tlsAdminHTTP.Config(s.logger) // This checks if the certificates are correctly provided
+	tlsAdminHTTPConfig := tlsAdminHTTP.ToOtelServerConfig()
+	if tlsAdminHTTPConfig != nil {
+		s.serverCfg.TLSSetting = tlsAdminHTTPConfig
+		_, err = s.serverCfg.TLSSetting.LoadTLSConfig(context.Background())
 		if err != nil {
 			return err
 		}
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:89]:
```suggestion
```

File: cmd/internal/flags/admin.go [hunk 10]
```diff
@@ -115,34 +114,42 @@ func (s *AdminServer) Serve() error {
...
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
-		var err error
-		if s.tlsCfg != nil {
-			err = s.server.ServeTLS(l, "", "")
-		} else {
-			err = s.server.Serve(l)
-		}
+		defer wg.Done()
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:141]:
```suggestion
		wg.Done()
```

File: cmd/internal/flags/admin.go [hunk 11]
```diff
@@ -115,34 +114,42 @@ func (s *AdminServer) Serve() error {
...
+		err := s.server.Serve(l)
 		if err != nil && !errors.Is(err, http.ErrServerClosed) {
 			s.logger.Error("failed to serve", zap.Error(err))
 			s.hc.Set(healthcheck.Broken)
 		}
 	}()
+
+	go func() {
+		wg.Wait()
+	}()
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:151]:
```suggestion
	wg.Wait()
```

File: cmd/internal/flags/admin.go [hunk 12]
```diff
@@ -101,12 +98,14 @@ func (s *AdminServer) Handle(path string, handler http.Handler) {
...
-	l, err := net.Listen("tcp", s.adminHostPort)
+	l, err := s.serverCfg.ToListener(context.Background())
 	if err != nil {
 		s.logger.Error("Admin server failed to listen", zap.Error(err))
 		return err
 	}
-	s.serveWithListener(l)
+	if err = s.serveWithListener(l); err != nil {
+		return err
+	}
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:108]:
```suggestion
```

File: cmd/internal/flags/admin.go [hunk 13]
```diff
@@ -115,34 +114,42 @@ func (s *AdminServer) Serve() error {
 	return nil
 }
```
Comment (yurishkuro) [inline cmd/internal/flags/admin.go:116]:
```suggestion
```
